### PR TITLE
Rebuild icu69

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/qt-feedstock/pull/207
Closes https://github.com/conda-forge/qt-feedstock/pull/212
Closes https://github.com/conda-forge/qt-feedstock/pull/213

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -46,6 +46,7 @@ source:
       - patches/0013-qtwebengine-missing_EVIOCGPROP.diff
       - patches/0014-qtwebengine-HAVE_SENDMMSG.diff
       - patches/google_breakpad_aarch_include.patch
+      - patches/0015-qtwebengine-fix-bison-gt-36.patch
```
Comment (hmaarrfk) [inline recipe/meta.yaml:49]: oh nice.

**Comment by akrherz** [comment]
This is impressive and hard work you folks are putting in, but I am curious of the goals here given what has happened with the new `qt-main` repo.  Could somebody tersely summarize what is happening ?  Perhaps I am the only one curious :)

**Comment by hmaarrfk** [comment]
Maybe this thread can help https://github.com/conda-forge/qt-feedstock/issues/192

I remember somebody announcing it on a thread somewhere, but I'm not sure where.

The goal here is mostly to migrate this until something else is fixed between qt and qt-main (and qt-webengine).

Ultimately, I think the `qt` package will just depend on `qt-main` and `qt-webengine` but I'm not sure when that will happen.

**Comment by hmaarrfk** [comment]
@izahn how did you hack the aarch64 build again? The hack seems to require python 2 which is not in the mian channel for aarch64.

**Comment by izahn** [comment]
> @izahn how did you hack the aarch64 build again? The hack seems to require python 2 which is not in the mian channel for aarch64.

Sorry which hack? I might have done something but I have an old brain and don't remember.

**Comment by izahn** [comment]
> This is impressive and hard work you folks are putting in, but I am curious of the goals here given what has happened with the new `qt-main` repo. Could somebody tersely summarize what is happening ? Perhaps I am the only one curious :)

The goal is to keep `qt` in sync with the rest of the conda-forge ecosystem by merging the icu69 migration PR. Keeping conda-forge packages in sync with the global pinning is the mechanism by which packages are kept compatible with each other. Because the icu69 migration PR has been languishing qt is currently incompatible with the most recent `tensorflow-base` and everything else that has been migrated to `icu 69`

**Comment by akrherz** [comment]
> The goal is to keep `qt` in sync with the rest of the conda-forge ecosystem

Thank you and totally agree as I have been thrashing about a lot within conda-forge due to the missing qt + icu69 combination :)  This would be awesome to figure out!

**Comment by hmaarrfk** [comment]
@izahn was it you that cared about aarch64? maybe it was somebody else. I'm trying to use the conda-forge channel for the hack instead of main:
https://github.com/conda-forge/qt-feedstock/pull/213/commits/51b448515364d6c54ad36733476b8a80a9ed2e2c

Lets see if that works.

**Comment by izahn** [comment]
> @izahn was it you that cared about aarch64? maybe it was somebody else. I'm trying to use the conda-forge channel for the hack instead of main: [51b4485](https://github.com/conda-forge/qt-feedstock/commit/51b448515364d6c54ad36733476b8a80a9ed2e2c)

That is horrifying :-)

**Comment by hmaarrfk** [comment]
what makes you think we no longer need to merge build and host for qtwebengine?

**Comment by izahn** [comment]
> what makes you think we no longer need to merge build and host for qtwebengine?

`python build-locally.py` completed the build without error.

**Comment by hmaarrfk** [comment]
oh. very cool!

**Comment by hmaarrfk** [comment]
Do we have time to try removing the openssl hack? I feel like `openssl` is already installed in the host environment....

**Comment by izahn** [comment]
> Do we have time to try removing the openssl hack? I feel like `openssl` is already installed in the host environment....

My motivation for improvements here is limited because https://github.com/conda-forge/qt-main-feedstock will soon replace it if all goes well. I think we should just get the icu migration out the door and call it a day.

**Comment by hmaarrfk** [comment]
I guess you should at least include the fix for finding python-2 and openssl from conda-forge instead of main. Using mamba will help the solve time a bit too.

**Comment by izahn** [comment]
I feel optimistic that what we have here works and is good enough. I'm building `linux_64_` now, any volunteers to build the others? We need

- [ ] linux_aarch64_
- [ ] osx_64_
- [ ] win_64_

**Comment by hmaarrfk** [comment]
I can't seem to access the conda forge aarch server.

@Tobias-Fischer do you have an other machine?

**Comment by Tobias-Fischer** [comment]
No, I was building on the conda-forge aarch server ..

**Comment by izahn** [comment]
`linux_64_` build is at https://anaconda.org/izahn/qt/5.12.9/download/linux-64/qt-5.12.9-ha98a1a1_5.tar.bz2 and the log file is at https://www.dropbox.com/s/enwa7l4oegbcdu3/qt_linux_64_.log?dl=0

**Comment by Tobias-Fischer** [comment]
In principle I am happy to build on aarch (this is what I promised), however 1) I think the PR should first be properly reviewed so we don't end up building multiple times, and 2) I'll be on leave till 4 January ..

**Comment by izahn** [comment]
@conda-forge/qt this is ready for review.

**Comment by hmaarrfk** [comment]
Webkit seems to be failing on aarch:
```
[2396/16095] CXX obj/skia/skia_opts_crc32/SkOpts_crc32.o
FAILED: obj/skia/skia_opts_crc32/SkOpts_crc32.o
$SRC_DIR/g++ -MMD -MF obj/skia/skia_opts_crc32/SkOpts_crc32.o.d -DV8_DEPRECATION_WARNINGS -DUSE_UDEV -DUSE_AURA=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DNO_TCMALLOC -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DOFFI
CIAL_BUILD -DCHROMIUM_BUILD -DFIELDTRIAL_TESTING_ENABLED -DTOOLKIT_QT -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DSK_IGNORE_LINEONLY_AA_CONVEX_PATH_OPTS -DS
K_HAS_PNG_LIBRARY -DSK_HAS_WEBP_LIBRARY -DSK_HAS_JPEG_LIBRARY -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.h\" -DSK_GAMMA_EXPONENT=1.2 -DSK_GAMMA_CONTRAST=0.2 -DSK_DEFAULT_FONT_CACHE_
LIMIT=20971520 -Igen -I../../3rdparty/chromium -I../../3rdparty/chromium/skia/config -I../../3rdparty/chromium/skia/ext -I../../3rdparty/chromium/third_party/skia/include/c -I../../3rdparty/chromium/third_party/skia/include/config -
I../../3rdparty/chromium/third_party/skia/include/core -I../../3rdparty/chromium/third_party/skia/include/effects -I../../3rdparty/chromium/third_party/skia/include/encode -I../../3rdparty/chromium/third_party/skia/include/gpu -I../
../3rdparty/chromium/third_party/skia/include/images -I../../3rdparty/chromium/third_party/skia/include/lazy -I../../3rdparty/chromium/third_party/skia/include/pathops -I../../3rdparty/chromium/third_party/skia/include/pdf -I../../3
rdparty/chromium/third_party/skia/include/pipe -I../../3rdparty/chromium/third_party/skia/include/ports -I../../3rdparty/chromium/third_party/skia/include/utils -I../../3rdparty/chromium/third_party/skia/src/gpu -I../../3rdparty/chr
omium/third_party/skia/src/sksl -I../../3rdparty/chromium/third_party/skia/include/codec -I../../3rdparty/chromium/third_party/skia/include/private -I../../3rdparty/chromium/third_party/skia/include/client/android -I../../3rdparty/c
hromium/third_party/skia/src/codec -I../../3rdparty/chromium/third_party/skia/src/core -I../../3rdparty/chromium/third_party/skia/src/image -I../../3rdparty/chromium/third_party/skia/src/images -I../../3rdparty/chromium/third_party/
skia/src/opts -I../../3rdparty/chromium/third_party/skia/src/pdf -I../../3rdparty/chromium/third_party/skia/src/ports -I../../3rdparty/chromium/third_party/skia/src/shaders -I../../3rdparty/chromium/third_party/skia/src/shaders/grad
ients -I../../3rdparty/chromium/third_party/skia/src/sfnt -I../../3rdparty/chromium/third_party/skia/src/utils -I../../3rdparty/chromium/third_party/skia/src/lazy -I../../3rdparty/chromium/third_party/skia/third_party/gif -I../../3r
dparty/chromium/third_party/skia/src/effects/gradients -march=armv8-a+crc -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -pipe -pthread -fno-omit-frame-pointer -g0 -fvisibility=hidden -O2 -fno
-ident -fdata-sections -ffunction-sections -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-deprecated-declarations -fno-delete-null-pointer-checks -Wno-comments -Wno-dangling-else -Wno-packed-not-aligned -Wno-missing-field-
initializers -Wno-unused-parameter -std=gnu++14 -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -Wno-narrowing -Wno-attributes -Wno-class-memaccess -Wno-subobject-linkage -c ../../3rdparty/chromium/third_party/skia/src/opts/Sk
Opts_crc32.cpp -o obj/skia/skia_opts_crc32/SkOpts_crc32.o
In file included from ../../3rdparty/chromium/third_party/skia/src/opts/SkOpts_crc32.cpp:11:
../../3rdparty/chromium/third_party/skia/src/opts/SkChecksum_opts.h:22:1: error: template with C linkage
   22 | template <typename T, typename P>
      | ^~~~~~~~
In file included from ../../3rdparty/chromium/third_party/skia/src/opts/SkChecksum_opts.h:17,
                 from ../../3rdparty/chromium/third_party/skia/src/opts/SkOpts_crc32.cpp:11:
$BUILD_PREFIX/lib/gcc/aarch64-conda-linux-gnu/9.4.0/include/arm_acle.h:33:1: note: 'extern "C"' linkage started here
   33 | extern "C" {
      | ^~~~~~~~~~
../../3rdparty/chromium/third_party/skia/src/opts/SkOpts_crc32.cpp:17:1: error: expected '}' at end of input
   17 | }
      | ^
In file included from ../../3rdparty/chromium/third_party/skia/src/opts/SkChecksum_opts.h:17,
                 from ../../3rdparty/chromium/third_party/skia/src/opts/SkOpts_crc32.cpp:11:
$BUILD_PREFIX/lib/gcc/aarch64-conda-linux-gnu/9.4.0/include/arm_acle.h:33:12: note: to match this '{'
   33 | extern "C" {
      |            ^
[2397/16095] CXX obj/skia/skia_core_and_effects/SkXfermodeImageFilter.o
ninja: build stopped: subcommand failed.
make[4]: Leaving directory '$SRC_DIR/qtwebengine/src/core'
make[3]: Leaving directory '$SRC_DIR/qtwebengine/src/core'
```

**Comment by izahn** [comment]
Hmm, looks like the CI got stuck and never got to that point: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=429108&view=logs&jobId=6494c7a6-a019-59a3-6f3d-9bab8ca96717&j=6494c7a6-a019-59a3-6f3d-9bab8ca96717&t=47f1ac7e-d0be-5d8d-6da0-bd921ed9abfb

@conda-foge-admin please rerender

**Comment by hmaarrfk** [comment]
I don't think you'll get to that point on Azure. It uses qemu and is very slow.

**Comment by izahn** [comment]
> @conda-forge/qt this is ready for review.

I was wrong, still some things to work out. Will ping you again when this is ready.

**Comment by Tobias-Fischer** [comment]
Possibly related: 
- https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100985
- https://github.com/xmrig/xmrig/issues/2596
- https://tracker.ceph.com/issues/51147

**Comment by hmaarrfk** [comment]
i guess we can try to downgrade the compiler?

**Comment by izahn** [comment]
@hmaarrfk and @Tobias-Fischer I added you to my forked repo in case you want to push anything directly there. Or we can continue with https://github.com/conda-forge/qt-feedstock/pull/213 if that's easier. I have no preference for which PR gets merged, I just want to get this migration taken care of so I can install the latest tensorflow and qt in the same environment.

**Comment by hmaarrfk** [comment]
the thing is, nothing changed in terms of the compiler version for Aarch. so I'm not sure

**Comment by hmaarrfk** [comment]
I think that this package should be moved forward ignoring the aarch failures. This issue should be able to be dealt with separately.

The main issue, opencv + qt, will be addressed here. 

For opencv, we don't build with qt support for aarch (yet) so I think it will be safe to ignore.

https://github.com/conda-forge/opencv-feedstock/blob/master/recipe/meta.yaml#L94

**Comment by hmaarrfk** [comment]
I'm not sure who can build osx, or windows.

**Comment by Tobias-Fischer** [comment]
I'm happy to give the aarch build a try - are the patches etc all fine?

**Comment by hmaarrfk** [comment]
All the links above refer to gcc patches. I'm not sure where to apply those in conda-forge.

**Comment by Tobias-Fischer** [comment]
Oh, sorry for the confusing question. I was referring to the status of the PR, making sure that everything else looks good. I'll try aarch with a downgraded compiler.

**Comment by hmaarrfk** [comment]
I mean, the current state compiles on linux it seems.

**Comment by Tobias-Fischer** [comment]
I've built the win and osx packages:
- Win package: https://anaconda.org/TobiasRobotics/qt/5.12.9/download/win-64/qt-5.12.9-h556501e_5.tar.bz2
- Win log: https://www.dropbox.com/s/sldnfrtgi7d7nh5/build_log_win.txt?dl=0
- Osx package: https://anaconda.org/TobiasRobotics/qt/5.12.9/download/osx-64/qt-5.12.9-h2a607e2_5.tar.bz2
- Osx log: https://www.dropbox.com/s/abaxn8b5qj0d6za/build_log_osx.log?dl=0

I currently don't have access to the conda-forge aarch server, and thus cannot built the aarch package. As soon as I have access, I am happy to try building the package with gcc11 (rather than 9).

**Comment by Tobias-Fischer** [comment]
Actually gcc 9.3 should work - it looks like the bug was introduced in gcc 9.4. The last aarch build was done with gcc 9.3.

@hmaarrfk - would you be happy to try?

**Comment by hmaarrfk** [comment]
@Tobias-Fischer i would be happy to trigger the build. I don't have much time to write the yml file that will pin to 9.3 for Aarch. could you do that and rerender and push? i can trigger after that.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
I started the rebuild on aarch. the rerendering shows that the other builds (not aarch64) were not affected. So hopefully all should be ready in 1 day.

**Comment by hmaarrfk** [comment]
I'm not really sure how to get the environment to solve. if you can get that working with an older compiler i can launch it.

**Comment by Tobias-Fischer** [comment]
It seems like indeed for libgcc=9.3 and icu=69 the environment is impossible to solve. Is there any objection in pinning the compilers version to 11? In this case I can solve locally (without the additional nspr/sqlite pins).

@izahn - could you please re-invite me to your fork?

**Comment by hmaarrfk** [comment]
i don't think I'm against going to gcc 11.

**Comment by izahn** [comment]
> It seems like indeed for libgcc=9.3 and icu=69 the environment is impossible to solve. Is there any objection in pinning the compilers version to 11? In this case I can solve locally (without the additional nspr/sqlite pins).
> 
> @izahn - could you please re-invite me to your fork?

@Tobias-Fischer I just sent the invite again, sorry for the delay. Happy new year!

**Comment by Tobias-Fischer** [comment]
Okay, I got the PR to build on aarch. At the moment the tests fail because some system requirements (libGL etc) are not installed; I asked Isuru to install them and will run the build again once that's done.

To get the PR to build, I needed to tweak some bits:
- add `git` and `patch` to the build reqs as they are not installed on the aarch server
- add a new patch (https://github.com/conda-forge/qt-feedstock/blob/ef79ff74f81de172ca1761eb882028b3fa251096/recipe/patches/0016-fix-gcc-11.patch) that fixes some missing includes which gcc11 complained about. In theory we could include this patch on all platforms, but to avoid rebuilding the other platforms I added a selector for `linux and aarch64`.

**Comment by izahn** [comment]
Has there been any movement here @Tobias-Fischer @hmaarrfk ? I think we just need to build and upload right?

**Comment by Tobias-Fischer** [comment]
Hi @isuruf - sorry to ping you. Could you please install the libGL system dependencies on the aarch server so I can build again with passing tests?

Thanks!

**Comment by isuruf** [comment]
Done

**Comment by Tobias-Fischer** [comment]
Thanks - I triggered the aarch build.

**Comment by Tobias-Fischer** [comment]
Aarch package: https://anaconda.org/TobiasRobotics/qt/5.12.9/download/linux-aarch64/qt-5.12.9-hb57a898_5.tar.bz2
Aarch log: https://www.dropbox.com/s/pelomc1k725go3z/build_log_aarch.txt?dl=0

**Comment by pkgw** [comment]
I just want to say: huge thanks to everyone working hard on these builds! The frustrating thing about this kind of work is that the better job you do, the less that the rest of the world even gets to appreciate how challenging it is ...

File: recipe/build.sh [hunk 2]
```diff
@@ -63,7 +63,8 @@ rm -rf $PREFIX/bin/protoc
 # Then find the .gn or .gni files that these ninja files were created from and figure out wtf is going on.
 
 # qtwebengine needs python 2
-conda create -y --prefix "${SRC_DIR}/python2_hack" -c https://repo.continuum.io/pkgs/main --no-deps python=2
+# conda create -y --prefix "${SRC_DIR}/python2_hack" -c https://repo.continuum.io/pkgs/main --no-deps python=2
```
Comment (hmaarrfk) [inline recipe/build.sh:66]: Its probably not worth rebuilding with this line removed, but maybe we can remember to remove it in the future.

Comment (Tobias-Fischer) [reply recipe/build.sh:66]: It's already commented out, isn't it?

**Comment by Tobias-Fischer** [comment]
I think we have all builds now. Next step is someone from core to look at the logs and upload the packages to the conda-forge channel?

**Comment by hmaarrfk** [comment]
@conda-forge/core could you please upload the packages

**Comment by izahn** [comment]
@Tobias-Fischer I don't understand why but `mamba` can't seem to find the windows build:

```
CONDA_SUBDIR=win-64 mamba create -n test qt=5.12.9=h556501e_5 -q --dry-run
Encountered problems while solving:
  - nothing provides requested qt ==5.12.9 h556501e_5
```
And `mamba` wants to install the old one instead:
```
ONDA_SUBDIR=win-64 mamba create -n test qt=5.12.9=h556501e_5 -q --dry-run
Encountered problems while solving:
  - nothing provides requested qt ==5.12.9 h556501e_5

 ~ [1] $ CONDA_SUBDIR=win-64 mamba create -n test qt -q --dry-run
  Package                 Version  Build               Channel                  Size
──────────────────────────────────────────────────────────────────────────────────────
  Install:
──────────────────────────────────────────────────────────────────────────────────────

  + ca-certificates     2021.10.8  h5b45459_0          conda-forge/win-64     176 KB
  + icu                      68.2  h0e60522_0          conda-forge/win-64      16 MB
  + jpeg                       9e  h8ffe710_0          conda-forge/win-64     373 KB
  + libclang               11.1.0  default_h5c34c98_1  conda-forge/win-64      21 MB
  + libpng                 1.6.37  h1d00b33_2          conda-forge/win-64     724 KB
  + libzlib                1.2.11  h8ffe710_1013       conda-forge/win-64      63 KB
  + openssl                1.1.1l  h8ffe710_0          conda-forge/win-64       6 MB
  + qt                     5.12.9  h5909a2a_4          conda-forge/win-64     106 MB
  + sqlite                 3.37.0  h8ffe710_0          conda-forge/win-64       1 MB
  + ucrt             10.0.20348.0  h57928b3_0          conda-forge/win-64       1 MB
  + vc                       14.2  hb210afc_6          conda-forge/win-64      13 KB
  + vs2015_runtime    14.29.30037  h902a5da_6          conda-forge/win-64       1 MB
  + zlib                   1.2.11  h8ffe710_1013       conda-forge/win-64     106 KB
```
Any idea what is going wrong here?

**Comment by izahn** [comment]
I'm especially confused because https://anaconda.org/conda-forge/qt/5.12.9/download/win-64/qt-5.12.9-h556501e_5.tar.bz2 exists, but somehow `mamba` doesn't see it.

**Comment by jakirkham** [comment]
Probably a sync issue. Working with Anaconda to resolve

**Comment by isuruf** [comment]
@izahn, please open issues in the future instead of commenting on PRs already merged.

https://github.com/conda/infra/discussions/514

**Pull request merged**